tests/integration_tests/security_tests.py

```diff
@@ -1718,14 +1719,28 @@ def test_raise_for_access_rbac(
         world_health = self.get_dash_by_slug("world_health")
         treemap = self.get_slice("Treemap", db.session, expunge_from_session=False)
 
+        births.json_metadata = json.dumps(
+            {
+                "native_filter_configuration": [
+                    {
+                        "id": "NATIVE_FILTER-ABCDEFGH",
+                        "targets": [{"datasetId": birth_names.id}],
+                    },
+                    {
+                        "id": "NATIVE_FILTER-IJKLMNOP",
+                        "targets": [{"datasetId": treemap.id}],
+                    },
+                ]
+            }
+        )
+
         mock_g.user = security_manager.find_user("gamma")
         mock_is_owner.return_value = False
         mock_can_access.return_value = False
         mock_can_access_schema.return_value = False
 
         for kwarg in ["query_context", "viz"]:
             births.roles = []
-            db.session.flush()
```

Comment:
No need to flush these updates to the database transaction given the logic in the test is confined to the same Flask-SQLAlchemy session.